helix-core/src/main/java/org/apache/helix/controller/stages/AttributeName.java

```diff
@@ -48,5 +48,7 @@
   // This attribute should only be used in TaskGarbageCollectionStage, misuse could cause race conditions.
   TO_BE_PURGED_WORKFLOWS,
   // This attribute should only be used in TaskGarbageCollectionStage, misuse could cause race conditions.
+  JOBS_WITHOUT_CONFIG,
```

Comment:
One minor question, why not call it TO_BE_PURGED_JOB directly?

Reply:
I used the name you suggested at the beginning. Junkai asked for the change and I think it makes sense to have a new name. TO_BE_PURGED_JOB causes confusion between jobs without config and the expired job that have passed their expiry time.